File: camerax_ios/lib/src/video/quality_selector_impl.dart
// import 'package:camerax_ios/src/camerax_api.g.dart';
// import 'package:camerax_ios/src/common.dart';
// import 'package:camerax_ios/src/core.dart';
// import 'package:camerax_platform_interface/camerax_platform_interface.dart';

// import 'fallback_strategy_impl.dart';
// import 'quality_impl.dart';

// final class QualitySelectorImpl extends QualitySelector {
//   static Future<Size<int>?> getResolution(
//     CameraInfo cameraInfo,
//     Quality quality,
//   ) => QualitySelectorProxyApi.getResolution(
//     cameraInfo.api,
//     quality.api,
//   ).then((e) => e?.impl);

//   final QualitySelectorProxyApi api;

//   QualitySelectorImpl.internal(this.api) : super.impl();

//   factory QualitySelectorImpl.from(
//     Quality quality, {
//     FallbackStrategy? fallbackStrategy,
//   }) {
//     final api = fallbackStrategy == null
//         ? QualitySelectorProxyApi.from1(quality: quality.api)
//         : QualitySelectorProxyApi.from2(
//             quality: quality.api,
//             fallbackStrategy: fallbackStrategy.api,
//           );
//     return QualitySelectorImpl.internal(api);
//   }

//   factory QualitySelectorImpl.fromOrderedList(
//     List<Quality> qualities, {
//     FallbackStrategy? fallbackStrategy,
//   }) {
//     final api = fallbackStrategy == null
//         ? QualitySelectorProxyApi.fromOrderedList1(
//             qualities: qualities.map((e) => e.api).toList(),
//           )
//         : QualitySelectorProxyApi.fromOrderedList2(
//             qualities: qualities.map((e) => e.api).toList(),
//             fallbackStrategy: fallbackStrategy.api,
//           );
//     return QualitySelectorImpl.internal(api);
//   }
// }

// extension QualitySelectorX on QualitySelector {
//   QualitySelectorProxyApi get api {
//     final impl = this;
//     if (impl is! QualitySelectorImpl) throw TypeError();
//     return impl.api;
//   }
// }

// extension QualitySelectorProxyApiX on QualitySelectorProxyApi {
//   QualitySelector get impl => QualitySelectorImpl.internal(this);
// }

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/video/QualitySelectorImpl.kt
package dev.hebei.camerax_android.video

import android.util.Size
import androidx.camera.core.CameraInfo
import androidx.camera.video.FallbackStrategy
import androidx.camera.video.Quality
import androidx.camera.video.QualitySelector
import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiQualitySelectorApi

class QualitySelectorImpl(impl: CameraXImpl) : PigeonApiQualitySelectorApi(impl) {
    override fun from(quality: Quality, fallbackStrategy: FallbackStrategy?): QualitySelector {
        return if (fallbackStrategy == null) QualitySelector.from(quality)
        else QualitySelector.from(quality, fallbackStrategy)
    }

    override fun fromOrderedList(qualities: List<Quality>, fallbackStrategy: FallbackStrategy?): QualitySelector {
        return if (fallbackStrategy == null) QualitySelector.fromOrderedList(qualities)
        else QualitySelector.fromOrderedList(qualities, fallbackStrategy)
    }

    override fun getResolution(cameraInfo: CameraInfo, quality: Quality): Size? {
        return QualitySelector.getResolution(cameraInfo, quality)
    }
}